src/main/java/com/powsybl/openloadflow/network/util/GenerationActivePowerDistributionStep.java

```diff
@@ -43,7 +43,7 @@ public String getElementType() {
     @Override
     public List<ParticipatingElement> getParticipatingElements(Collection<LfBus> buses) {
         return buses.stream()
-                .filter(bus -> !(bus.isDisabled() || bus.isFictitious()))
+                .filter(bus -> !(bus.isDisabled() || bus.isFictitious() || !bus.isParticipating()))
```

Comment:
The double negative is not well understandable...

Reply:
The alternative is bus.isParticipating() && !bus.isDisabled() && !bus.isFictitious(), I think its better, do you agree ?

Reply:
Yes :-)